server/routes/getting_started/getting_started_router.ts

```diff
@@ -134,4 +129,75 @@ export function registerGettingStartedRoutes(router: IRouter) {
       }
     }
   );
+
+  router.post(
+    {
+      path: `/api/observability/gettingStarted/createAssets`,
+      validate: {
+        body: schema.object({
+          mdsId: schema.string(),
+          mdsLabel: schema.string(),
+          tutorialId: schema.string(),
+        }),
+      },
+    },
+    async (
+      context,
+      request,
+      response
+    ): Promise<IOpenSearchDashboardsResponse<any | ResponseError>> => {
+      try {
+        const { mdsId, mdsLabel, tutorialId } = request.body;
+        const fileData = await loadAssetsFromFile(tutorialId);
+
+        const objects = await createSavedObjectsStreamFromNdJson(Readable.from(fileData));
+        const loadedObjects = await objects.toArray();
+
+        const updatedObjects = loadedObjects.map((obj) => {
+          if (mdsId) {
+            const newId = `mds-${mdsId}-objectId-${obj.id}`;
+
+            const newReferences =
+              obj.references?.map((ref: { type: string; id: any }) => {
+                if (ref.type === 'visualization') {
+                  return {
+                    ...ref,
+                    id: `mds-${mdsId}-objectId-${ref.id}`,
+                  };
+                }
+                return ref;
+              }) || [];
+
+            newReferences.push({
+              id: mdsId,
+              type: 'data-source',
+              name: mdsLabel,
+            });
```

Comment:
Can you please validate with @sumukhswamy if references are needed for all the created saved objects. IIRC it is only needed for index patterns. Since, visualizations have a reference to index pattern and dashboards have a reference to visualizations we can infer the original datasource by tracking other references.

Reply:
After meeting with Sumukh only the index-pattern will have the added reference. The otheres will just get the updated ID's
<img width="557" alt="Screenshot 2024-08-16 at 9 34 57 AM" src="https://github.com/user-attachments/assets/4c6956db-d4b0-4826-8921-a137afa6b1b3">
<img width="856" alt="Screenshot 2024-08-16 at 9 34 47 AM" src="https://github.com/user-attachments/assets/f3fda37f-894d-4bf7-b43e-cc8c57f25e9b">